File: CS/DocumentMerger/Helpers/DocumentsMerger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraRichEdit;
using System.Data;
using DevExpress.XtraRichEdit.API.Native;

namespace DocumentMerger.Helpers {
    public class DocumentsMerger {

        public static Document MergeDouments(List<string> filenames) {
            RichEditDocumentServer targetServer = new RichEditDocumentServer();
            RichEditDocumentServer sourceServer = new RichEditDocumentServer();
            Document targetDoc = targetServer.Document;
            Document sourceDoc = sourceServer.Document;

            for (int i = 0; i < filenames.Count; i++) {
                sourceServer.LoadDocument(filenames[i]);

                targetDoc.Sections[targetDoc.Sections.Count - 1].UnlinkHeaderFromPrevious();
                targetDoc.Sections[targetDoc.Sections.Count - 1].UnlinkFooterFromPrevious();

                SectionsMerger.Append(sourceDoc, targetDoc);

                if (i == filenames.Count - 1)
                    return targetDoc;

                targetDoc.AppendSection();
            }

            return targetDoc;
        }
    }
}

File: VB/DocumentMerger/Helpers/DocumentsMerger.vb
Imports System.Collections.Generic
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.API.Native

Namespace DocumentMerger.Helpers

    Public Class DocumentsMerger

        Public Shared Function MergeDouments(ByVal filenames As List(Of String)) As Document
            Dim targetServer As RichEditDocumentServer = New RichEditDocumentServer()
            Dim sourceServer As RichEditDocumentServer = New RichEditDocumentServer()
            Dim targetDoc As Document = targetServer.Document
            Dim sourceDoc As Document = sourceServer.Document
            For i As Integer = 0 To filenames.Count - 1
                sourceServer.LoadDocument(filenames(i))
                targetDoc.Sections(targetDoc.Sections.Count - 1).UnlinkHeaderFromPrevious()
                targetDoc.Sections(targetDoc.Sections.Count - 1).UnlinkFooterFromPrevious()
                Call SectionsMerger.Append(sourceDoc, targetDoc)
                If i = filenames.Count - 1 Then Return targetDoc
                targetDoc.AppendSection()
            Next

            Return targetDoc
        End Function
    End Class
End Namespace
